app/gasprice/gasprice.go

```diff
@@ -0,0 +1,97 @@
+package gasprice
+
+import (
+	"math/big"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/params"
+
+	appconfig "github.com/okex/exchain/app/config"
+	"github.com/okex/exchain/app/types"
+)
+
+var (
+	maxPrice     = big.NewInt(500 * params.GWei)
+	defaultPrice = big.NewInt(params.GWei / 10)
+)
+
+type GPOConfig struct {
+	Weight  int
+	Default *big.Int `toml:",omitempty"`
+}
+
+func NewGPOConfig(weight int, defaultPrice *big.Int) GPOConfig {
+	return GPOConfig{
+		Weight:  weight,
+		Default: defaultPrice,
+	}
+}
+
+func DefaultGPOConfig() GPOConfig {
+	return NewGPOConfig(appconfig.GetOecConfig().GetDynamicGpWeight(), defaultPrice)
+}
+
+// Oracle recommends gas prices based on the content of recent blocks.
+type Oracle struct {
+	CurrentBlockGPs types.SingleBlockGPs
+	// hold the gas prices of the latest few blocks
+	BlockGPQueue types.BlockGPResults
+	lastPrice    *big.Int
+	weight       int
+}
+
+// NewOracle returns a new gasprice oracle which can recommend suitable
+// gasprice for newly created transaction.
+func NewOracle(params GPOConfig) *Oracle {
```

Comment:
It seems that GPOConfig cannot be configured via cli or toml, so I didn't find the benefit of GPOConfig, can we directly use NewOracle(weight, default) or NewDefaultOracle?

Reply:
Actually the "Weight" in GPOConfig is configured through cli, the default "Weight" is 80. "Default" is not currently supported, but I am worried that maybe we need to configure "Default" through cli in the future.